### DISCUSSION TITLE
Change language option

### DISCUSSION THREAD

**Discussion by ReptiIe** [PintaProject/Pinta#804] [Ideas]
I notice that app has translations, but there is no option to choose it? I guess app loads the same language as users os,
but it would be way better if users had the option to choose what language Pinta should be (some might be using OS in one language, but app they would prefer to have in their native language).

**Comment by cameronwhite** [comment]
Yeah it follows the OS setting, although it can be overridden by environment variables like `LANG`. 
If you're using the macOS the system language settings can be changed per app, but I'm not sure offhand how possible that is on other OS

**Reply by ReptiIe** [reply]
I am testing app from the Windows 11 os, Pinta 2.1.1
and I notice that users were asking for this option since year 2013 somewhere...
bit odd.

I mean somehow it is added, Pinta will adapt itself to system os, 
but it would be better to just place option to choose the language.


Edit: 20/05/2024
I had install Pinta 2.1.2, windows 11os, 
sure the problem still is the same. Just to make sure.

**Reply by M3n747** [reply]
I had the same issue and found your question while looking for an answer. I didn't find one, but I did find out that you can change the language quite simply, by going to `./Pinta/locale`, removing/renaming the directory with your current language and copying the directory with your preferred language in its place. So in my case, I renamed the `pl` directory to `pl-OLD` and then copied `en_GB` to `pl`, and it works perfectly fine.

(I know I'm pretty late with this one, but perhaps somebody in the future will also look to do the same.)

**Reply by ReptiIe** [reply]
Thanks for replay @M3n747 
There is no real answer to this. I mean yes, as a normal user you might (if you need to), make some tricks, play with files, change it etc etc., yes it is a excellent tip! 

But the thing is devs should think long time ago to implement this option, so the app would work normally as any other apps.
"Additional fighting" with files as a "normal user" it... a bit of a turn off what I mean.

**Reply by M3n747** [reply]
Well, no argument there - the option to switch languages should be there from the start. But at the very least there is a way to do that, even if it's not the most intuitive one. Fingers crossed the devs implement this before the end of the decade. :P

**Comment by BlueHtml** [comment]
Hello, I installed `Pinta 3.0` and I'm using `Windows 10 Simplified Chinese zh-CN`. But when I open the Pinta, it displays English. How can I switch the language to Chinese?

**Reply by ReptiIe** [reply]
@BlueHtml 
Can not. There is absolutely no option, function for that.  Like [M3n747](https://github.com/M3n747) from above mention you can open `./Pinta/locale` find your language folder, copy it and paste it in the `en_GB` I guess? 

Though, this also did not work for me.

as for today (29/04/2025), I had also install Pinta 3.0, but still there is no function to change language.
Have to wait or well try forcing it somehow like M3n747 said.

**Reply by BlueHtml** [reply]
@ReptiIe Thank you for your reply. I also tried that and it indeed doesn't work.

I renamed the `zh_CN` folder under `Pinta\share\locale` to `en`, but it didn't work. The software still displays in English when opened.

It's too hard for me to use. I've decided to uninstall it.

**Comment by brynne8** [comment]
Manually set the LANG variable in Win11 to zh_CN doesn't work either.

**Reply by ReptiIe** [reply]
yeah, well sadly all we can do is wait until a `normal option` for `choosing language` will be added.
There is nothing much we can do. I am already waiting like more than 3 years, and year ago I decided to drop this suggestion...

**Reply by cameronwhite** [reply]
Please file a bug / feature request for this (https://github.com/PintaProject/Pinta/issues) - the discussions section is not a bug tracker

**Reply by ReptiIe** [reply]
@cameronwhite ah yes, I had started this topic in a... hmmm... "IDEAS" sanction, so I believe this was good.
It seams I should mark it as a BUG. My bad. 
Anyway I started a new topic if that helps.

**Reply by cameronwhite** [reply]
No problem, this was the right place for discussing ideas around having more options for language selection, but later comments have veered more into issues where it sounds like an unexpected language is showing up by default

**Comment by xanathon** [comment]
I just wanted to chime in that I find it highly ridiculous that one has to fiddle with lang folders or conf file variables for something simple as changin the application language. This should be in a preferences entry in the menu.

**Comment by Mario-Hofstaetter** [comment]
I just installed pinta for the first time only to discover there is no option to change the language. Or rather, there is no configuration option at all in the GUI? What in the World?
Sadly, will uninstall immediately.

**Reply by ReptiIe** [reply]
Yup. I mention this issue in 2024, right now it will be 2026, 
this issue still remain.

And I am still waiting :-)

**Reply by Mario-Hofstaetter** [reply]
No option to choose language when you have multiple translations is an absurdity to me. Usually this cancer is only plaguing me on android apps.

**Comment by SPrite075** [comment]
You can change language remove everything in pinta>share>locale but left only the language you want

**Reply by ReptiIe** [reply]
Nope, you can not. I just tried with the newest version.

Still even IF it would work on your PC... how the F## , 99% of normal users would change a language in a app like that.
By deleting files...
This is crazy. 

May 2026 - still waiting for a normal **option** or **preferences** menu that you will simply click in the app, 
and change the language to what you need... That it will work like in 99% of every app out there.

**Comment by xanathon** [comment]
I gave up on Pinta because of this. There are better alternatives out there that have language selection.

**Reply by Skif-off** [reply]
> There are better alternatives

For example?

**Reply by xanathon** [reply]
Krita

**Reply by ReptiIe** [reply]
Depends what you are looking for, but there are a lot of fish in the ocean as they say. Naming just few:
PixiEditor, Gimp, Krita, Pixelorama, Pixelitor, Aseprite, Libresprite, Lazpaint, Piskel


Every app has its own pros and cons, sure.
But example: if you are looking for an app that provides your language option (not to play with absurd methods like file swapping, editing, deleting as is mention here - but a normal language option to choose in the app itself) then there is not much of a brainer here.

**Reply by Skif-off** [reply]
Gimp or Krita is another level, sounds almost like Adobe Photoshop vs. Paint.NET.
LazPaint has good functionality, but also some kind of space interface.. but I am just a simple earthling :)
The other names sound familiar, but I can't remember, I'll have to look them up.

**Comment by SilverGreen93** [comment]
I cannot believe there is no option to change the UI language. In fact Pinta seems to not have any configurable options at all...

**Reply by ReptiIe** [reply]
Yes you're right. I am still waiting for this option since 2024 😄

**Comment by cameronwhite** [comment]
Closing this since #1417 is implemented now in the `master` branch
If you'd like to give it a test, https://github.com/PintaProject/Pinta/actions/runs/35052362480 has pre-built installers for various platforms
